pkg/network/netlink/conntrack.go

```diff
@@ -34,41 +36,59 @@ type Conntrack interface {
 // NewConntrack creates an implementation of the Conntrack interface.
 // `netNS` is the network namespace for the conntrack operations.
 // A value of `0` will use the current thread's network namespace
-func NewConntrack(netNS int) (Conntrack, error) {
-	conn, err := netlink.Dial(unix.NETLINK_NETFILTER, &netlink.Config{NetNS: netNS})
+func NewConntrack(netNS netns.NsHandle) (Conntrack, error) {
+	conn, err := NewSocket(netNS)
 	if err != nil {
 		return nil, err
 	}
-	return &conntrack{conn: conn}, nil
+
+	return &conntrack{
+		conn: conn,
+		msg: netlink.Message{
+			Header: netlink.Header{
+				Type:  netlink.HeaderType((unix.NFNL_SUBSYS_CTNETLINK << 8) | ipctnlMsgCtGet),
+				Flags: netlink.Request | netlink.Acknowledge,
+			},
+		},
+	}, nil
 }
 
 type conntrack struct {
-	conn *netlink.Conn
+	sync.Mutex
+	conn *Socket
+	msg  netlink.Message
```

Comment:
Can we document this? It's not clear why we'd have a single message in this struct

Reply:
Done